### PR TITLE
College Licence Auto-Adjudication Rule

### PR DESCRIPTION
**Pull request by james-hollinger**
Also some refactoring

### CODE REVIEW

File: prime-dotnet-webapi/Models/Certification.cs [hunk 1]
```diff
@@ -44,5 +44,9 @@ public class Certification : BaseAuditable, IEnrolleeNavigationProperty
 
         [JsonIgnore]
         public Practice Practice { get; set; }
+
+        [NotMapped]
+        [JsonIgnore]
+        public string FullLicenceNumber { get { return $"{College?.Prefix}-{LicenseNumber}"; } }
```
Comment (itchison) [inline prime-dotnet-webapi/Models/Certification.cs]: I feel there are some inconsistencies with licence vs license in our code.

File: prime-dotnet-webapi/Services/DefaultEnrolleeService.cs [hunk 2]
```diff
@@ -318,109 +318,89 @@ public Task<EnrolmentStatus> CreateEnrolmentStatusAsync(int enrolleeId, Status s
...
+
+                case Status.DECLINED_TOS_CODE:
+                    await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
+                    createdEnrolmentStatus.PharmaNetStatus = true;
+                    break;
+            }
 
-                return createdEnrolmentStatus;
+            var created = await _context.SaveChangesAsync();
+            if (created < 1)
```
Comment (itchison) [inline prime-dotnet-webapi/Services/DefaultEnrolleeService.cs]: This reads weird doesn't it?What is greater than 1?

Comment (james-hollinger) [reply prime-dotnet-webapi/Services/DefaultEnrolleeService.cs]: this returns the number of records created when saving the database. It's existing code that has moved in the file

File: prime-dotnet-webapi/Models/Certification.cs (hunk 1 above)
Comment (james-hollinger) [reply prime-dotnet-webapi/Models/Certification.cs]: I agree, I defaulted to the "correct" spelling of licence for Canada here without thinking. I feel like there are 3 options:

1. Change this to FullLicenseNumber to be consistent, if wrong
2. Change LicenseNumber to LicenceNumber on the Certification only
3. Change every instance of License to Licence (where grammatically appropriate)

This PR already has a fair bit of refactoring in it, so whats a little bit more? On the other hand, I would like to get this feature in, it's a high profile one. I'm leaning towards 2.

Comment (james-hollinger) [reply prime-dotnet-webapi/Models/Certification.cs]: I have done #1 for consistency

**Pull request merged**